### PR TITLE
Add SphinxTranslator as an abstract class

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Refactoring

### Purpose
- To help annotation, and to make translators simple, I propose to Introduce `SphinxTranslator` class

### CODE REVIEW

**Comment by tk0miya** [comment]
@cocoatomo what do you think about this?

File: sphinx/writers/latex.py [hunk 1]
```diff
@@ -505,8 +507,7 @@ class LaTeXTranslator(nodes.NodeVisitor):
 
     def __init__(self, document, builder):
         # type: (nodes.document, LaTeXBuilder) -> None
-        super(LaTeXTranslator, self).__init__(document)
-        self.builder = builder
+        super(LaTeXTranslator, self).__init__(builder, document)
```
Comment (tk0miya) [inline sphinx/writers/latex.py:514]: Unfortunately, the order of arguments are different between Translators.
I think it's time to integrate to one way.

File: sphinx/util/docutils.py [hunk 2]
```diff
@@ -344,6 +345,32 @@ def config(self):
         return self.env.config
 
 
+class SphinxTranslator(nodes.NodeVisitor):
```
Comment (tk0miya) [inline sphinx/util/docutils.py:387]: Is there any appropriate place for this class? `util` package is not best to me, but I don't understand where is better.

File: sphinx/writers/latex.py (hunk 1 above)
Comment (tk0miya) [reply sphinx/writers/latex.py:514]: It must be a breaking change. So I'll post this as another PR.

**Review by jfbu — COMMENTED** [review]
Since this PR, in LaTeX output for document with no language settings one sees

    \addto\captionsenglish{\renewcommand{\contentsname}{None}}

with "None" in place of expected "Contents". Ping @tk0miya 

(or rather than Contents, there simply should *not* be any such line in LaTeX document)

**Pull request merged**